app/scripts/metamask-controller.js

```diff
@@ -1271,6 +1271,7 @@ export default class MetamaskController extends EventEmitter {
         oldAccounts.concat(accounts.map((a) => a.address.toLowerCase())),
       ),
     ];
+    this.preferencesController.syncAddresses(accountsToTrack);
```

Comment:
I don't think we want this :thinking: The accounts here are being previewed during the hardware connect process. This includes the current "page" that you're looking at. We don't want to add identities for each of these.